tools/fork-network/src/cli.rs

```diff
@@ -0,0 +1,285 @@
+use std::path::{Path, PathBuf};
+
+use near_chain::types::Tip;
+use near_chain_configs::{Genesis, GenesisConfig, GenesisRecords, GenesisValidationMode};
+use near_epoch_manager::{EpochManager, EpochManagerAdapter};
+use near_primitives::{
+    account::Account, borsh::BorshSerialize, hash::CryptoHash, types::AccountInfo,
+};
+use near_store::{flat::FlatStorageStatus, DBCol, Mode, NodeStorage, HEAD_KEY};
+use nearcore::{load_config, NightshadeRuntime};
+
+use crate::storage_mutator::{self, StorageMutator};
+
+#[derive(clap::Parser)]
+pub struct ForkNetworkCommand {
+    #[arg(short, long)]
+    pub reset: bool,
+    #[arg(short, long, default_value = "1000")]
+    pub epoch_length: u64,
+}
+
+impl ForkNetworkCommand {
+    pub fn run(
+        self,
+        home_dir: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        if self.reset {
```

Comment:
I like this idea, it's cleaner than hacking the migration snapshot, but I think I'll need to try it later as a lower priority item. The problem is that I not only need to remove columns, but also need to write additional state. I don't wanna write that to the current db, and I also cannot remove columns and *then* write because then I won't be able to instantiate important data structures. So, I would have to checkpoint, change the config and *reopen* the checkpoint db, then change the data, and then delete columns. And that makes it not feel so clean anymore.